components/Document/DocumentViewer.tsx

```diff
@@ -300,6 +322,43 @@ const DocumentViewer = ({
   const shouldScroll =
     actualContentWidth > windowDimensions.width - LEFT_SIDEBAR_MAX_WIDTH;
 
+  const uploadPDF = (acceptedFiles: File[] | any[]): void => {
```

Comment:
The function itself is sound. Ideally, I'd like to have it one level up (in whatever is referencing DocumentViewer) so that `DocumentViewer` isn't aware or concerned with references. That way we can it more pure and also reduce the number of props, mainly removing:

  setReferenceItemDatum,
  referenceItemDatum,